src/actions/experiments/create.rs

```diff
@@ -47,39 +99,45 @@ impl Action for CreateExperiment {
-                 VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12);",
-                &[
-                    &self.name,
-                    &self.mode.to_str(),
-                    &self.cap_lints.to_str(),
-                    &self.toolchains[0].to_string(),
-                    &self.toolchains[1].to_string(),
-                    &self.priority,
-                    &Utc::now(),
-                    &Status::Queued.to_str(),
-                    &self.github_issue.as_ref().map(|i| i.api_url.as_str()),
-                    &self.github_issue.as_ref().map(|i| i.html_url.as_str()),
-                    &self.github_issue.as_ref().map(|i| i.number),
-                    &self.ignore_blacklist,
-                ],
-            )?;
-
-            for krate in &crates {
-                let skipped = !self.ignore_blacklist && ctx.config.should_skip(krate);
-                transaction.execute(
-                    "INSERT INTO experiment_crates (experiment, crate, skipped) VALUES (?1, ?2, ?3);",
-                    &[&self.name, &::serde_json::to_string(&krate)?, &skipped],
-                )?;
-            }
+                  transaction.execute(
+                      "INSERT INTO experiments \
+                      (name, mode, cap_lints, toolchain_start, toolchain_end, priority, created_at, \
+                      status, github_issue, github_issue_url, github_issue_number, children, ignore_blacklist) \
+                      VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12, ?13);",
+                      &[
+                          &self.name,
+                          &self.mode.to_str(),
+                          &self.cap_lints.to_str(),
+                          &self.toolchains[0].to_string(),
+                          &self.toolchains[1].to_string(),
+                          &self.priority,
+                          &Utc::now(),
+                          &Status::Queued.to_str(),
+                          &self.github_issue.as_ref().map(|i| i.api_url.as_str()),
+                          &self.github_issue.as_ref().map(|i| i.html_url.as_str()),
+                          &self.github_issue.as_ref().map(|i| i.number),
+                          &((crates.len() / CHUNK_SIZE + 1) as i32),
+                          &self.ignore_blacklist,
+                      ],
+                  )?;
+
+                  for krate in &crates {
+                      let skipped = !self.ignore_blacklist && ctx.config.should_skip(krate);
+                      transaction.execute(
+                          "INSERT INTO experiment_crates (experiment, crate, skipped) VALUES (?1, ?2, ?3);",
+                          &[&self.name, &::serde_json::to_string(&krate)?, &skipped],
+                      )?;
+                  }
 
-            Ok(())
-        })?;
+                  Ok(())
+              })?;
+
+        for i in 0..=crates.len() / CHUNK_SIZE {
```

Comment:
You can use `for chunk in crates.chunks(CHUNK_SIZE)` to avoid having to use `get_portion`.

Reply:
Also, I could save crates in `experiment_crates` with the name of the chunk instead of the experiment and recreate an `ExperimentChunk` with all the records with given chunk name